cpp/include/cudf/reduction/detail/reduction_functions.hpp

```diff
@@ -131,6 +131,37 @@ std::unique_ptr<scalar> all(column_view const& col,
                             rmm::cuda_stream_view stream,
                             rmm::mr::device_memory_resource* mr);
 
+/**
+ * @brief Compute frequency for each unique element in the input column.
+ *
+ * The result histogram is stored in structs column having two children. The first child contains
+ * unique elements from the input, and the second child contains their corresponding frequencies.
+ *
+ * @throw cudf::logic_error if `output_dtype` is not integer type
+ *
+ * @param input The column to compute histogram
+ * @param output_dtype Data type to store the element frequencies
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return A list_scalar storing a structs column as the result histogram
```

Comment:
So the output is a `list_scalar` and the elements in the list are structs?

Reply:
Yes. Indeed, the output should be a structs column, but reduction can only output a scalar so the structs column is placed in a list. This is also consistent with the output of groupby, where the output is a lists column of structs column, each list corresponds to the output histogram of one key.